src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlTransactionHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.SqlServer.Configs;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
+{
+    internal class SqlTransactionHandler : ITransactionHandler
+    {
+        private readonly SqlServerDataStoreConfiguration _configuration;
+
+        public SqlTransactionHandler(SqlServerDataStoreConfiguration configuration)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            _configuration = configuration;
+        }
+
+        public SqlTransactionScope SqlTransactionScope { get; private set; }
+
+        public ITransactionScope BeginTransaction()
+        {
+            Debug.Assert(SqlTransactionScope == null, "The existing SQL transaction scope should be completed before starting a new transaction.");
```

Comment:
Should this throw exception and abort the current transaction if it actually happens? Would we have obscure bug by letter the new transaction scope in?

Reply:
Yes, we should just throw here.